solarforecastarbiter/reports/template.py

```diff
@@ -16,179 +14,48 @@
 logger = logging.getLogger(__name__)
 
 
-def template_report(report, metadata, metrics,
-                    processed_forecasts_observations):
-    """
-    Render the markdown report template. Figures are left untemplated.
-
-    Parameters
-    ----------
-    report : solarforecastarbiter.datamodel.Report
-        Metadata describing report
-    metadata : solarforecastarbiter.datamodel.ReportMetadata
-        Describes the pre-report
-    metrics : tuple of dict
-    processed_forecasts_observations : tuple of solarforecastarbiter.datamodel.ProcessedForecastObservation
-
-    Returns
-    -------
-    markdown
-    """  # noqa
-    # By default, jinja removes undefined variables from the rendered string.
-    # DebugUndefined leaves undefined variables in the string so that they
-    # can be used in the full report template process.
+def _get_render_kwargs(report, dash_url, with_timeseries):
+    kwargs = dict(
+        human_categories=datamodel.ALLOWED_CATEGORIES,
+        human_metrics=datamodel.ALLOWED_METRICS,
+        report=report,
+        dash_url=dash_url,
+        bokeh_version=report.raw_report.plots.bokeh_version
+    )
+    if with_timeseries:
+        try:
+            script, div = figures.timeseries_plots(report)
+        except Exception:
+            logger.exception(
+                'Failed to make Bokeh items for timeseries and scatterplot')
+            script = ''
+            div = """<div class="alert alert-warning">
+  <strong>Warning</strong> Failed to make timeseries and scatter plots
+  from stored data. Try generating report again.
+</div>
+            """
+        kwargs['timeseries_script'] = script
+        kwargs['timeseries_div'] = div
+    return kwargs
```

Comment:
should be able to update/extend report by adding kwargs. Should make sure that any additions have defaults/provide a warning to not break reports made with older versions